### PR TITLE
Hit Registration Fix

### PR DESCRIPTION
**Pull request by ampreeT**
This PR fixes a issue with hit registration being off by 1 tick.

**Changes:**

- Added hook for Black Mesa and HL2DM to fix hit registration while moving

### CODE REVIEW

File: scripting/include/srccoop/playerpatch.inc [hunk 1]
```diff
@@ -6,6 +6,8 @@
 #pragma newdecls required
 #pragma semicolon 1
 
+static float g_pvec3PlayerShootPosition[MAXPLAYERS][3];
```
Comment (Alienmario) [inline scripting/include/srccoop/playerpatch.inc:9]: [MAXPLAYERS + 1] and probably unintentional p in variable name

File: scripting/include/srccoop/playerpatch.inc [hunk 2]
```diff
@@ -612,6 +615,23 @@ public Action Hook_PlayerTakeDamage(int victim, int &attacker, int &inflictor, f
 	return Plugin_Continue;
 }
 
+// Fixes hit registration being off by 1 tick.
+// Credits to Xutax_Kamay (https://forums.alliedmods.net/showthread.php?t=315405)
+public MRESReturn Hook_PlayerWeaponShootPosition_Post(int iClient, Handle hParams)
```
Comment (Alienmario) [inline scripting/include/srccoop/playerpatch.inc:620]: hParams can be removed if there are none

**Review by Alienmario — CHANGES_REQUESTED** [review]

File: gamedata/srccoop.games.txt [hunk 3]
```diff
@@ -1808,6 +1820,11 @@
 				"windows"	"13"
 				"linux"		"13"
 			}
+			"CBasePlayer::Weapon_ShootPosition"	// CBasePlayer::Weapon_ShootPosition()
```
[the remark this answers is not in the document]
Comment (ampreeT) [reply gamedata/srccoop.games.txt:1823]: Fixed the offsets.

File: scripting/include/srccoop/playerpatch.inc (hunk 2 above)
Comment (ampreeT) [reply scripting/include/srccoop/playerpatch.inc]: Renamed to `hReturn`.

File: scripting/include/srccoop/playerpatch.inc (hunk 1 above)
Comment (ampreeT) [reply scripting/include/srccoop/playerpatch.inc]: Yeah that was unintentional. Fixed the array size and removed `p` from the variable name.

**Review by Alienmario — APPROVED** [review]

**Pull request merged**